### DISCUSSION TITLE
SearXNG as a site search (self developed  public websites (php)

### DISCUSSION THREAD

**Discussion by spamhater-grav** [searxng/searxng#1424] [Q&A] [answered]
Just watched a video from networkchuck about setting up searxng, and a little bit of curve ball, but he stated about creating a local index, which got me thinking, could searxng but adapted or used to provide a site search function on self development website, similar to algolia without the overhead costs.

If so is there any examples or tutorials of this could be accomplished. 
many thank

**Comment by return42** [comment, marked as answer]
> If so is there any examples or tutorials of this could be accomplished.

In our [source code documentation](https://docs.searxng.org/src/index.html) we have examples for different engine types .. I assume you are looking for the [Demo Online Engine](https://docs.searxng.org/dev/engines/demo/demo_online.html).

----

Indexing is not a task of SearXNG, it is better covered by projects like [YaCy](https://en.wikipedia.org/wiki/YaCy) for what we have a [engine](https://github.com/searxng/searxng/blob/master/searx/engines/yacy.py).

https://github.com/searxng/searxng/blob/4ef62637256d4b2966ff7f0be0051e02d545f756/searx/settings.yml#L1677-L1684

**Reply by spamhater-grav** [reply]
Many thanks for taking out the time to reply. Great, I hopefully can take your info and attempt integration between both of the project. :)
